Fix coding rules errors

There are errors when running

```sh
$ ./mwnv clean test
[ERROR] Tests run: 6, Failures: 2, Errors: 1, Skipped: 0, Time elapsed: 0.061 s <<< FAILURE! - in org.eclipse.sirius.web.services.architecture.ImmutableTests
[ERROR] immutableClassesShouldHaveAPrivateConstructor  Time elapsed: 0.022 s  <<< FAILURE!
java.lang.AssertionError:
Architecture Violation [Priority: MEDIUM] - Rule 'classes that reside in a package 'org.eclipse.sirius.web.services..' and are annotated with @Immutable and are not annotated with @GraphQLInputObjectType should have only private constructors' was violated (2 times):
Constructor <org.eclipse.sirius.web.services.api.projects.UnzippedProject.<init>()> is not private in (UnzippedProject.java:31)
Constructor <org.eclipse.sirius.web.services.api.representations.RepresentationDescriptor.<init>()> is not private in (RepresentationDescriptor.java:27)

[ERROR] immutableClassesShouldBeFinal  Time elapsed: 0.005 s  <<< FAILURE!
java.lang.AssertionError:
Architecture Violation [Priority: MEDIUM] - Rule 'classes that reside in a package 'org.eclipse.sirius.web.services..' and are annotated with @Immutable should have modifier FINAL' was violated (2 times):
Class <org.eclipse.sirius.web.services.api.projects.UnzippedProject> does not have modifier FINAL in (UnzippedProject.java:0)
Class <org.eclipse.sirius.web.services.api.representations.RepresentationDescriptor> does not have modifier FINAL in (RepresentationDescriptor.java:0)

[ERROR] immutableClassesShouldHaveANestedBuilder  Time elapsed: 0.007 s  <<< ERROR!
java.lang.IllegalArgumentException: No field with name 'projectManifest in class org.eclipse.sirius.web.services.api.projects.UnzippedProject$Builder
```

They seem to appear in the code from the Sirius Web repository.

--- a/backend/sirius-web-services-api/src/main/java/org/eclipse/sirius/web/services/api/projects/UnzippedProject.java
+++ b/backend/sirius-web-services-api/src/main/java/org/eclipse/sirius/web/services/api/projects/UnzippedProject.java
@@ -28,7 +28,7 @@ import org.eclipse.sirius.web.spring.graphql.api.UploadFile;
  * @author gcoutable
  */
 @Immutable
-public class UnzippedProject {
+public final class UnzippedProject {
 
     private String projectName;
 
@@ -36,7 +36,11 @@ public class UnzippedProject {
 
     private Map<String, UploadFile> documentIdToUploadFile = new HashMap<>();
 
-    private ProjectManifest manifest;
+    private ProjectManifest projectManifest;
+
+    private UnzippedProject() {
+        // Prevent instantiation
+    }
 
     public String getProjectName() {
         return this.projectName;
@@ -50,8 +54,8 @@ public class UnzippedProject {
         return this.documentIdToUploadFile;
     }
 
-    public ProjectManifest getManifest() {
-        return this.manifest;
+    public ProjectManifest getProjectManifest() {
+        return this.projectManifest;
     }
 
     public static Builder newUnzippedProject(String projectName) {
@@ -71,7 +75,7 @@ public class UnzippedProject {
 
         private Map<String, UploadFile> documentIdToUploadFile;
 
-        private ProjectManifest manifest;
+        private ProjectManifest projectManifest;
 
         private Builder(String projectName) {
             this.projectName = Objects.requireNonNull(projectName);
@@ -88,7 +92,7 @@ public class UnzippedProject {
         }
 
         public Builder projectManifest(ProjectManifest manifest) {
-            this.manifest = Objects.requireNonNull(manifest);
+            this.projectManifest = Objects.requireNonNull(manifest);
             return this;
         }
 
@@ -97,7 +101,7 @@ public class UnzippedProject {
             unzippedProject.projectName = Objects.requireNonNull(this.projectName);
             unzippedProject.representationDescriptors = Objects.requireNonNull(this.representationDescriptors);
             unzippedProject.documentIdToUploadFile = Objects.requireNonNull(this.documentIdToUploadFile);
-            unzippedProject.manifest = Objects.requireNonNull(this.manifest);
+            unzippedProject.projectManifest = Objects.requireNonNull(this.projectManifest);
             return unzippedProject;
         }
     }

--- a/backend/sirius-web-services-api/src/main/java/org/eclipse/sirius/web/services/api/representations/RepresentationDescriptor.java
+++ b/backend/sirius-web-services-api/src/main/java/org/eclipse/sirius/web/services/api/representations/RepresentationDescriptor.java
@@ -24,7 +24,7 @@ import org.eclipse.sirius.web.representations.IRepresentation;
  * @author gcoutable
  */
 @Immutable
-public class RepresentationDescriptor {
+public final class RepresentationDescriptor {
     private UUID id;
 
     private UUID projectId;
@@ -36,6 +36,10 @@ public class RepresentationDescriptor {
     private String label;
 
     private IRepresentation representation;
+
+    private RepresentationDescriptor() {
+        // Prevent instantiation
+    }
 
     public UUID getId() {
         return this.id;

--- a/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/projects/ProjectImportService.java
+++ b/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/projects/ProjectImportService.java
@@ -93,7 +93,7 @@ public class ProjectImportService implements IProjectImportService {
             return new ErrorPayload(inputId, this.messageService.unexpectedError());
         }
         UnzippedProject unzippedProject = optionalUnzippedProject.get();
-        ProjectManifest manifest = unzippedProject.getManifest();
+        ProjectManifest manifest = unzippedProject.getProjectManifest();
         String projectName = unzippedProject.getProjectName();
 
         CreateProjectInput createProjectInput = new CreateProjectInput(inputId, projectName, Visibility.PRIVATE);

